File: src/array_string/largestNumber.js
/**
 * @param {number[]} nums
 * @return {string}
 */
// var largestNumber = function(nums) {

//     nums.map(x=>x.toString())
//      console.log("--- "+nums);
//     let cmp = (a,b) => {
//         console.log("a .. "+a+" b.. "+b);
        
//         return a+b>b+a ? 1:-1}
    
//     nums.sort(cmp)
//     console.log(nums);
    
//     result = nums.join("")
//     if(result[0]==="0")
//       return "0"
//     return result;
// };
// test = [3,30,34,5,9]  // 9534330
// console.log(largestNumber(test)
// );


function largestNumber(nums) {
    // Convert the numbers to strings
    let s_nums = nums.map(x=>x.toString())
    // nums.map(String);

    // Custom comparator function
    function comp(a, b) {
        let s1 = a + b;
        let s2 = b + a;
        
        // Sorting in descending order based on the comparison of concatenated strings
        return s1 > s2 ? -1 : 1;
    }

    // Sort the array based on the custom comparator
    s_nums.sort(comp);

    console.log("give the temp: ", s_nums);

    // Join the sorted numbers into a single string
    let result = s_nums.join('');

    // Handle the edge case where the result starts with '0' (e.g., [0, 0])
    return result[0] === '0' ? '0' : result;
}

// Example usage
console.log(largestNumber([3, 30, 34, 5, 9]));

File: src/array_string/largestNumber.py
from functools import cmp_to_key


def largestNumber(nums):
        # convert and split nums int single dig in form charactor
               
     s_nums = [str(s) for s in nums ]
     print("source: ",s_nums)
     def comp(a,b): 
        print("a: ",a, "-- b: ",b)
        s1 =a+b
        s2= b+a
        
        return -1 if s1>s2 else 1
     s_nums.sort(key=cmp_to_key(comp))
     print("give the temp: ",s_nums)
     result="".join(s_nums)
     return 0 if result[0]==0 else result

test = [3,30,34,5,9]  # 9534330

print(largestNumber(test))


# def largestNumber(self, nums: List[int]) -> str:
#         s=[str(x) for x in nums]
#         def cmp(x, y):
#             return -1 if x+y>y+x else 1
#         s.sort(key=cmp_to_key(cmp))
#         ans="".join(s)
#         return '0' if ans[0]=='0' else ans
        
        